controllers/openstackmachine_controller.go

```diff
@@ -426,6 +434,36 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 	}
 	conditions.MarkTrue(openStackMachine, infrav1.APIServerIngressReadyCondition)
 
+	// Apply floating ip on worker if enabled in openStackCluster spec
+	if !util.IsControlPlaneMachine(machine) && openStackCluster.Spec.WorkerFloatingIPConfig.Enabled {
+		scope.Logger().Info("Processing worker floating IPs")
+		for _, floatingIP := range openStackCluster.Spec.WorkerFloatingIPConfig.IPAddresses {
+			fp, err := networkingService.GetOrCreateFloatingIP(openStackMachine, openStackCluster, clusterName, floatingIP)
+			if err != nil {
+				conditions.MarkFalse(openStackMachine, infrav1.InstanceReadyCondition, infrav1.FloatingIPErrorReason, clusterv1.ConditionSeverityError, "Floating IP cannot be obtained or created: %v", err)
+				return ctrl.Result{}, fmt.Errorf("get or create floating IP %q: %w", floatingIP, err)
+			}
+
+			port, err := computeService.GetManagementPort(openStackCluster, instanceStatus)
+			if err != nil {
+				conditions.MarkFalse(openStackMachine, infrav1.InstanceReadyCondition, infrav1.FloatingIPErrorReason, clusterv1.ConditionSeverityError, "Obtaining management port for worker machine failed: %v", err)
+				return ctrl.Result{}, fmt.Errorf("get management port for worker machine: %w", err)
+			}
+
+			if fp.PortID != "" {
+				scope.Logger().Info("Floating IP already associated to a port", "id", fp.ID, "fixedIP", fp.FixedIP, "portID", port.ID)
+			} else {
+				err = networkingService.AssociateFloatingIP(openStackMachine, fp, port.ID)
+				if err != nil {
+					conditions.MarkFalse(openStackMachine, infrav1.InstanceReadyCondition, infrav1.FloatingIPErrorReason, clusterv1.ConditionSeverityError, "Associating floating IP failed: %v", err)
+					return ctrl.Result{}, fmt.Errorf("associate floating IP %q to worker node with port %q: %w", fp.FloatingIP, port.ID, err)
+				}
+			}
+		}
```

Comment:
We should log a situation when node haven't got any FIP because pool got exhausted.